### PR TITLE
FLPATH-861 Promote M2K workflow from examples to prod

### PR DESCRIPTION
**Pull request by dmartinol**
Adding GH action to build and push `m2k-func` functions according to the README file:

```
cd move2kubeAPI
make install
cd ..
mvn clean install
docker build -t quay.io/orchestrator/m2k-kfunc:2.0.0-SNAPSHOT -f src/main/docker/Dockerfile.jvm .
```

Required secrets added to the upstream repo according to the `orchestrator-ci` [robot account](https://quay.io/organization/orchestrator?tab=robots)

### CODE REVIEW

File: .github/workflows/m2k-func.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        cd ..
+        mvn ${MVN_OPTS} -B clean package -DskipTests
+
+    - name: Buildah Action
+      id: build-image
+      uses: redhat-actions/buildah-build@v2
+      with:
+        context: ${{ env.WORKDIR }}
+        image: serverless-workflow-m2k-kfunc
+        tags: latest ${{ github.sha }}
```
Comment (gabriel-farache) [inline .github/workflows/m2k-func.yaml:37]: this is the tag that will be used when pushing to the repo, correct? 
Just wondering as in the description of the PR you put `2.0.0-SNAPSHOT` but I guess that a simple copy paste

Comment (dmartinol) [reply .github/workflows/m2k-func.yaml:37]: Yes that was a copy and paste from README
For versioning I'd follow the logic used in the other production repo, if you agrere

Comment (gabriel-farache) [reply .github/workflows/m2k-func.yaml:37]: > For versioning I'd follow the logic used in the other production repo, if you agrere
Yes, I do, it's way better :)

**Review by gabriel-farache — APPROVED** [review]

**Review by masayag — APPROVED** [review]

**Pull request merged**